### DISCUSSION TITLE
OnQueryStarted dispatch mutation and can't get error in the component using hook

### DISCUSSION THREAD

**Discussion by JoseAntpr** [reduxjs/redux-toolkit#2606] [Q&A]
Hi community, recently I´ve started to use RTK Query for a personal PoC thinking to use in the company project.

In my proof of concept I´m using two injectEndpoint apis and I have a login in two steps. 

CODE ....

```ts
login: build.mutation<loginResponse, loginArgs>({
      queryFn: async ({ email, password }) => {
        const result = await loginWithEmailAndPassword({ email, password });

        const { login, password: plexPassword } = (await (
          await getDoc(doc(FibaseDB, "example", result.uid))
        ).data()) as { login: string; password: string };
        return { data: { ...result, login, password: password2} };
      },
      onQueryStarted: async (_, { dispatch, queryFulfilled }) => {
        const {
          data: { email, uid, login, password },
        } = await queryFulfilled;

// I know that setAuth dispatch maybe its better to use addMatcher to performance
        dispatch(authSlice.actions.setAuth({ uid, email }));

// I want to dispatch this second mutation after the firtst login is fullfilled 
        dispatch(plexApi.endpoints.step2Login.initiate({ login, password }));
      },
    }),
```

The problem is when this second step of the login failed I want to show the error in the loginComponent but the hook doesn't work. 

```ts
export const LoginPage = () => {
  const [login, { error }] = useLoginMutation();
  const [_, { isError, data }] = useStep2LoginMutation();

....
}
```

isError is always false although mutation fails ... Only works if the triggers is done in the component but if the mutation fails done by onQueryStarted I can't get error or data using the hook.

Do we have another way to get the error in the component ? What's wrong ?

**Comment by phryneas** [comment]
If you call `useMutation` on the same endpoint in two different components, you get two independent results - and it's the same if you call `initiate` in one place and `useMutation` in the other. If you don't want to have that behaviour, please read up on [shared mutation results](https://redux-toolkit.js.org/rtk-query/usage/mutations#shared-mutation-results)

**Reply by JoseAntpr** [reply]
ohh sounds great but how can pass this fixedCacheKey  to the initiate ?

**Reply by JoseAntpr** [reply]
Ohh I see ... I can pass this option to initiate function too ... I´m going to test it this behaviour I´ll come back !!
